pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -607,10 +611,32 @@ message ExecRequest {
 }
 
 message ExecResponse {
-    // Streaming stdout
-    optional bytes stdout = 1;
-    // Streaming stderr
-    optional bytes stderr = 2;
+    // The fully qualified URL of the exec streaming server
+    optional string url = 1;
+}
+
+message AttachRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // Streaming stdin
+    optional bytes stdin = 2;
```

Comment:
> That's reasonable, but `bytes` indicates a streaming behavior while this is not a streaming API.

Oh, thanks, I missed that.

> And I think `bool tty` should also be added to `AttachRequest`

It was excluded from Attach since it needs to be specified when the container starts. See discussion [here](https://docs.google.com/document/d/1OE_QoInPlVCK9rMAx9aybRmgFiVjHpJCHI9LrfdNM_s/edit?disco=AAAAAysm47s#heading=h.1c8p2nyag4wv)